src/types.ts

```diff
@@ -61,10 +66,9 @@ interface Rect {
     x: number;
     y: number;
     w: number;
+    isInterval?: boolean;
```

Comment:
I think it's redundant. We can understand if a node is an interval from the presence of the `intervals` field.